### DISCUSSION TITLE
Inserting a blank page after each PDF page

### DISCUSSION THREAD

**Discussion by lassepe** [xournalpp/xournalpp#3317] [Q&A] [answered]
Thank you for this great software! I've been using it a lot years now and it has worked great for me. However, I do have one specific workflow which I'd like to automate a bit more:

When I review or read a paper, I typically set the view to "pair pages" and then add a blank pages for my own notes after every PDF page. This is already possible but it is a bit of tedious task  because it requires 4N mouse clicks for a N page document. Is there a way to insert multiple blank pages or even a direct action to interleave all pages with a blank page?

**Comment by LittleHuba** [comment, marked as answer]
Hi!
This is a task you can accomplish with a Lua Plugin. If I'm not mistaken we have a Wiki entry or a page on our homepage on our API for plugins. There are also some example plugins referenced, you may take inspiration from.

**Reply by rolandlo** [reply]
For reference, this is the page on our homepage: https://xournalpp.github.io/guide/plugins/plugins/

**Reply by Kentoseth** [reply]
The discussed plugin cannot be found here: https://xournalpp.github.io/guide/plugins/notable/

Unless it was implied that a plugin be built for this.

**Reply by rolandlo** [reply]
Yes, a plugin has still to be built. Here is a starter (for `main.lua`). Adjust it to your needs.
```lua
-- Register all Toolbar actions and intialize all UI stuff
function initUi()
  app.registerUi({["menu"] = "add blank pages", ["callback"] = "run", ["accelerator"] = "<Alt><Shift>a"});
end

-- Callback if the menu item is executed
function run()
  local docStructure = app.getDocumentStructure()
  local numPages = #docStructure["pages"]
  local p = 0
  for i = 1, numPages do
    p = p+1
    app.setCurrentPage(p)
    if docStructure["pages"][i]["pageTypeFormat"] == ":pdf" then
      app.uiAction({["action"] = "ACTION_NEW_PAGE_AFTER"})
      app.changeCurrentPageBackground("plain")
      p = p+1
    end
  end
end
```

**Comment by KronosTheLate** [comment]
+1 that this would be lovely to have
